wezterm-gui/src/termwindow/selection.rs

```diff
@@ -58,29 +63,42 @@ impl super::TermWindow {
     ) {
         self.selection(pane.pane_id()).seqno = pane.get_current_seqno();
         let mode = mode.unwrap_or(SelectionMode::Cell);
-        let (x, y) = match self.pane_state(pane.pane_id()).mouse_terminal_coords {
-            Some(coords) => coords,
+        let (x, y, position) = match self.pane_state(pane.pane_id()).mouse_terminal_coords {
+            Some(coords) => (coords.0.column, coords.1, coords.0),
             None => return,
         };
         match mode {
             SelectionMode::Cell => {
-                let end = SelectionCoordinate { x, y };
-                let selection_range = self.selection(pane.pane_id()).range.take();
-                let sel = match selection_range {
-                    None => {
-                        SelectionRange::start(self.selection(pane.pane_id()).start.unwrap_or(end))
-                            .extend(end)
-                    }
-                    Some(sel) => sel.extend(end),
+                let origin = self
+                    .selection(pane.pane_id())
+                    .origin
+                    .unwrap_or(SelectionCoordinate { x, y });
+                self.selection(pane.pane_id()).origin = Some(origin);
+
+                let (start_x, end_x) = if (origin.x <= x && origin.y == y) || origin.y < y {
```

Comment:
Could you add some comments around this code to explain this logic?

Reply:
Yeah, indeed it looks a bit esoteric without context 😂